sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -496,9 +688,19 @@ public Boolean exists() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Boolean> existsWithResponse(Duration timeout, Context context) {
-        Mono<Response<Boolean>> response = client.existsWithResponse(context);
-
-        return blockWithOptionalTimeout(response, timeout);
+        try {
+            Callable<Response<Void>> operation = () -> this.azureBlobStorage.getBlobs()
+                .getPropertiesNoCustomHeadersWithResponse(containerName, blobName, snapshot, versionId, null, null,
+                    null, null, null, null, null, null, customerProvidedKey, context);
+            return new SimpleResponse<>(sendRequest(operation, timeout, BlobStorageException.class), true);
+        } catch (RuntimeException e) {
+            if (ModelHelper.checkBlobDoesNotExistStatusCode(e) && e instanceof HttpResponseException) {
```

Comment:
This is also missing a check for `BlobErrorCode.BLOB_USES_CUSTOMER_SPECIFIED_ENCRYPTION` that is being checked in `BlobAsyncClientBase`. This is the reason why all the live tests are failing.